README.md

```diff
@@ -1,57 +1,422 @@
+1. Find the key `TenantId` and replace the existing value with your Azure AD tenant ID.
+1. Find the key `ClientId` and replace the existing value with the application ID (clientId) of `TodoListClient-authContext-webapp` app copied from the Azure portal.
+1. Find the key `ClientSecret` and replace the existing value with the key you saved during the creation of `TodoListClient-authContext-webapp` copied from the Azure portal.
+
+## Running the sample
+
+> For Visual Studio Users
+>
+> Clean the solution, rebuild the solution, and run it.  You might want to go into the solution properties and set both projects as startup projects, with the service project starting first.
+
+```console
+    cd TodoListClient-authContext-webapp
+    dotnet run
+```
+
+## Explore the sample
+
+### Configure the Auth context
+
+1. We'd first replicate the experience of an admin configuring the auth contexts. For that, browse to `https://localhost:44321` and sign-in using a tenant Admin account. Click on the **Admin** link on the menu.
+
+    ![Overview](./ReadmeFiles/Admin.png)
+2. As a first step, you will ensure that a set of Auth Context is already available in this tenant. Click the **CreateOrFetch** button to check if they exist. If they don't , the code will create three sample auth context entries for you. These three entires are named `Require strong authentication`, `Require compliant devices` and `Require trusted locations`.
+
+    > Note: The Graph permission, **Policy.ReadWrite.ConditionalAccess** is required for creating new auth context records. In production, the permission, **Policy.Read.ConditionalAccess** should be sufficient to read existing values and is thus the only one recommended.
+
+    ![Overview](./ReadmeFiles/Create-Fetch_Click.png)
+
+    Select an operation in the Todo list controller and map it with an `Authentication Context` value. To apply click **SaveOrUpdate**. This updates this mapping in the local app's database.
+
+>Note: When changing auth context mappings, have the user sign-out and sign-back in for the changes to take effect.
+
+3. Go to **Current config** link to get details of data saved on the Web API side in its database. You can  **Delete** if you need to delete a mapping from the local database.
+
+    ![Overview](./ReadmeFiles/ViewDetails.png)
+
+The web app is now ready to challenge users for step-up auth for the selected operations.
+
+### Configure a Conditional Access policy to use auth context in Azure portal
+
+1. Navigate to Azure Active Directory> Security > Conditional Access
+1. Select **New policy** and go to **Cloud apps or actions**. In dropdown select **Authentication context**. The newly created auth context values will be listed for you to be used in this CA policy.
+
+    ![Overview](./ReadmeFiles/AuthContext.png)
+
+    Select the value and create the policy as required. For example, you might want the user to satisfy a MFA challenge if the auth context value is 'Medium'.
```

Comment:
After few hours of MFA not kicking off, I've noticed that I have "0 users and groups selected". Not sure if a person should  know that, but it took time for me to see it.

Reply:
ya Ca policy setup can be funny sometimes